Add Docker support to Travis configuration for projects

Andrew-Chen-Wang opened the pull request:
[//]: # (Thank you for helping us out: your efforts mean great deal to the project and the community as a whole!)

[//]: # (Before you proceed:)

[//]: # (1. Make sure to add yourself to `CONTRIBUTORS.rst` through this PR provided you're contributing here for the first time)
[//]: # (2. Don't forget to update the `docs/` presuming others would benefit from a concise description of whatever that you're proposing)


## Description

[//]: # (What's it you're proposing?)
Add Docker support to the Travis configuration.

Edits allowed by maintainers meaning you can add commits to this patch, too.


## Rationale

[//]: # (Why does the project need that?)
Builds automatically fail on first push


## Use case(s) / visualization(s)

[//]: # ("Better to see something once than to hear about it a thousand times.")
No more fails on first push...

Andrew-Chen-Wang commented:
Failures usually arise with the lack of a database URL, but... it seems like this needs more of a fork branch than a simple patch apparently.

browniebroke commented:
The patch looked pretty good, and it would solve #2100. I'm not sure I understand your comment, it seems to be about a different issue?

Andrew-Chen-Wang commented:
Whoops, I didn’t properly explain myself. The pre-gen test hook I believe ran into an error if you setup travis with those commands. My current configuration for Travis is a little different because if you were to only

`docker-compose build && docker-compose up`

Then you’d run into an error with celerybeat not having tables existing yet. I added entrypoint’s postgres_ready() function to beat’s start files and added a SQL statement to make sure its tables existed first before starting, but I was debating if that would take up too much costs and connection pool would increase substantially on a Blue/Green deployment for several clusters. You can take a look at https://github.com/Andrew-Chen-Wang/cookiecutter-django-ecs-deploy for what I did (also archiving it because I realized there’s probably no way ECS works with cookiecutter django... still looking).
